### ISSUE TITLE
Image Block: Selecting multiple image blocks should still offer link action button (in toolbar)

### ISSUE THREAD

**Issue by porg** [WordPress/gutenberg#53126]
## User goal

Select multiple images and link them to:
1. the same custom URL
2. or to same relative URL type
  a) their own Media File
  b) their own Media Attachment Page.

## Current workaround
- **Repeat the same click orgy for each image** — inefficient!
- **Use an image gallery** which has the **Link to**: _None | Media File | Attachment Page_ attribute applied to all its images.
  - But a gallery is often not appropriate, e.g. images spread over page with content inbetween, which you nevertheless all want to link in one step, after having them selected in the Document Overview (left side tree pane).

## Status Quo & Problem

![WordPress - Selecting multiple Image Block should still offer the link action button](https://github.com/WordPress/gutenberg/assets/737143/a28eeb95-47dd-4512-a0da-72fa51bdb1f3)

## Proposal

When multiple image blocks are selected the toolbar should still offer the link action with all sub-actions:
  - **Media File**: Each image then points to its media file.
    - In the Media Library or on an external URL, whatever it is.
    - This action should also work with image sets consisting of both types: Some will then point to their src / srcset file(s) in the media library, others to external URLs.
  - **Attachment Page**: Each image then points to its media attachment page.
  - **Custom URL**: The same URL for all images.

✅ All these are valid use cases (sorted by most likely first).

**Comment by porg** [comment]
**"Link to" action for the image block** was integrated into the toolbar in
- #16451

Pinging people which were involved:
-  @noisysocks @jasmussen @youknowriad
- possibly having used code by @jorgefilipecosta

**Comment by jasmussen** [comment]
I tend to agree, though I think the issue at hand here can be more generic than images and their link targets. 

(There may even be an issue about this already.)

Essentially if you select multiple blocks, _ideally_ both the block toolbar and the inspector can become smart enough to identify similar properties and surface them together so you can apply across. Not just links, but also alignments, colors, etc. Figma does this well with colors, for example, where you can select across a variety of different options and change one color to another across.

**Comment by porg** [comment]
@jasmussen I found the generic issue on multi-select:
- #49228
  - [Comment there](https://github.com/WordPress/gutenberg/issues/49228#issuecomment-1657931594) to extend the scope to handle not only the values in the inspector, but also the toolbar actions.
- Also: **Linking a whole block** (wrapping the essential element(s) into an HTML A element) could become a generic attribute/feature which could be used by any block (via opt-in or opt-out, to be discussed).
  - That certainly would need to be a separate issue, maybe already exists.
- But as both these are more general/wider discussions, I nevertheless made this concrete specific proposal, as it builds on an existing attribute (link) and an existing function (multi-selection of values).
  - But ofc it would create "code debt" in a specific block, which better should be available as a general/shared feature, given general best practices.
  - I don't know the Gutenberg guidelines in that regard.

**Comment by jasmussen** [comment]
Individual actionable items can be useful if they cross-link the relevant higher level context 👍 

> Also: Linking a whole block (wrapping the essential element(s) into an HTML A element) could become a generic attribute/feature which could be used by any block (via opt-in or opt-out, to be discussed).

I can't speak with too much confidence as I've only touched this superficially, but I think this one may be challenging on multi-block selection due not just to the semantic nature of HTML which allow only certain tags inside the `a` anchor, but also due to some accessibility considerations when making bigger link buttons. To that end this one may end up being a single-block transform.

**Comment by porg** [comment]
- 👍 Thanks on your quick feedback that **Linking a whole block** as a generic feature has rather little chances, as
  - HTML A cannot take just any child element(s),
  - and as HTML & Modular CSS differs too much between different WordPress blocks,
    - which makes opt-out impossible (would likely simply break most blocks)
    - and even as opt-in, block authors may need to rework their CSS selectors, to also work as intended, when another A HTML gets inserted to their hierarchy, which not every block author will be willing.
  - 🙂 So well, that gives more justification to this very specific issue!
- ✅ Cross-linked this specific issue now to the higher level context issue.

**Comment by porg** [comment]
**Related use case** where the UI provides the necessary controls to change an attribute for multiple selected blocks of the same type but where the outcome is buggy:
- #53424
